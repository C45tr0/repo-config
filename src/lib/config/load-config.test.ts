import chdirSetup from "../../../test/helpers/chdir";

import loadConfig from "./load-config";

const chdir = chdirSetup("config");

test("returns default data if no config found", () => {
  chdir("no-config");

  expect(loadConfig()).toMatchSnapshot();
});

test("returns blank dir on top level config load", () => {
  chdir("package-only");

  expect(loadConfig().dir).toMatchSnapshot();
});

test("returns relative path on extenstion config load", () => {
  chdir("loads-extends");

  expect(loadConfig("node_modules/test-extend").dir).toMatchSnapshot();
});

test("throws error when custom config can't be found", () => {
  chdir("custom-config-not-found");

  expect(loadConfig).toThrowErrorMatchingSnapshot();
});

test("throws error when custom config can't be loaded", () => {
  chdir("custom-config-invalid");

  expect(loadConfig).toThrowErrorMatchingSnapshot();
});

test("throws error when repo.config.json can't be loaded", () => {
  chdir("config-invalid");

  expect(loadConfig).toThrowErrorMatchingSnapshot();
});
