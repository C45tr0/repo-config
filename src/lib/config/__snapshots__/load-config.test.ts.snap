// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns blank dir on top level config load 1`] = `""`;

exports[`returns default data if no config found 1`] = `
Object {
  "dir": "",
  "packageConf": Object {},
}
`;

exports[`returns relative path on extenstion config load 1`] = `"node_modules/test-extend"`;

exports[`throws error when custom config can't be found 1`] = `"Custom config file not found at path: custom.config.json"`;

exports[`throws error when custom config can't be loaded 1`] = `"Invalid custom config file: custom.config.json"`;

exports[`throws error when repo.config.json can't be loaded 1`] = `"Invalid config file: repo.config.json"`;
